src/VideoStreaming/VideoReceiver.cc

```diff
@@ -417,7 +396,7 @@ gboolean VideoReceiver::_onBusMessage(GstBus* bus, GstMessage* msg, gpointer dat
     Q_UNUSED(bus)
     Q_ASSERT(msg != NULL && data != NULL);
     VideoReceiver* pThis = (VideoReceiver*)data;
-    pThis->_onBusMessage(msg);
+    pThis->busMessage(msg);
```

Comment:
Ideally you'd actually do all the GstMessage processing here (what is currently done in _handleBusMessage) and use signals to perform specific actions on the VideoReceiver object (EOS handling, state change) by passing and necessary data as POD through the signal instead of the GstMessage pointer. 

Since you don't hold your own refcount to the message (you're called synchronously on the owner's thread) you can't ensure that it will live on after you return to the caller in the callback thread. When you emit a signal with the message pointer to a different thread, the pointer itself is queued and it becomes highly likely the underlying GstMessage pointed to is no longer valid by the time the slot gets invoked on the QObject's thread. 

You basically need two signals: msgEOSReceived() and msgStateChanged() that connect to slots that invoke the code in those sections of the switch statement. The GST_MESSAGE_ERROR block doesn't reference any state on your VideoReceiver object so that can go ahead and run in the callback thread context.